### ISSUE TITLE
Unable to connect, even after using port forwarding, switching to private connection. Very confused

### ISSUE THREAD

**Issue by magnumforce2006** [gregchak/plex-playlist-import#14]
Hey there @gregchak  I'm at a loss as far as what I'm doing wrong here. I'm including a screenshot of my settings, as well as the event log for the app. I've tried using both private and public addresses (pretty sure it's always supposed to be private?). And I've successfully forwarded the port via my router, and I've also tried after setting my server to private. Can't get anything to work. It's befuddling to me.

I have double checked and there's also an exception setup for Plex Playlist in my firewall as well.

Any help would be much appreciated!
[log20231231.txt](https://github.com/gregchak/plex-playlist-import/files/13802622/log20231231.txt)
![Plex_Playlist_setup](https://github.com/gregchak/plex-playlist-import/assets/9346053/97ccfd8f-9a3e-4952-9a9e-d73cb3a335f8)

**Comment by gregchak** [comment]
After looking at the logs it looks like the import tool is trying to connect to Plex at the URL specified (http://localhost:3240/) but it is unable to establish a connection.  This is typically the case when the URL is incorrect or there is something blocking it. In addition you may want to make sure that your Plex isn't set up for HTTPS connections only.  With the URL set to localhost I am assuming that you are running this import tool from the same computer that you are running Plex.  Is that correct?  You may want to check your firewall rules too to ensure that 32400 both inbound and outbound are allowed.  Since you are running Plex on this same computer I would assume 32400 is allowed inbound, but possibly not outbound, or possibly not in the right network type; i.e. private vs. public.  That's where I would start the troubleshooting.  No config is needed in your router since all of this local not only to your network but to your computer.

**Comment by magnumforce2006** [comment]
Yes I'm running this all on my computer which is where the files are located.

I've setup the outbound rule in my firewall. But still no luck. Am I setting the rule correctly? 
![Firewall screenshot](https://github.com/gregchak/plex-playlist-import/assets/9346053/3f42825d-a6e1-4fe0-a7a6-9be8b735b14c)

**Comment by gregchak** [comment]
See if you're able to hit the API directly from a browser:
`http://localhost:3240/servers?X-Plex-Token=[Your token]`

**Comment by magnumforce2006** [comment]
Nope, unable to connect to that. To be clear I'm inputting that without the brackets, yes?

**Comment by magnumforce2006** [comment]
![PlexCopy1](https://github.com/gregchak/plex-playlist-import/assets/9346053/48a5379b-c3c3-42c5-b816-5f7aefd1f9e3)
![PlexCopy2](https://github.com/gregchak/plex-playlist-import/assets/9346053/2c6f446a-e387-4b4e-83ca-d0b4bbd9f7bf)

Just out of curiosity I turned off the firewall and tried, still didn't work... Let's make sure I'm getting my token correctly I guess? I have no idea how I'd be screwing that up. Attached is a screenshot of what I'm copying and pasting, with the xml file I'm pulling and then the link.

**Comment by magnumforce2006** [comment]
I get a blank screen when inputting. Is that expected?
![PlexScreen1](https://github.com/gregchak/plex-playlist-import/assets/9346053/a9abdf7c-dc39-4cbe-a754-86b18c664fa5)

**Comment by magnumforce2006** [comment]
In Chrome this is what I get when inputting into address bar:
![PlexScreenChrome](https://github.com/gregchak/plex-playlist-import/assets/9346053/ee9799c4-38de-4706-9dd5-954f3eea9d81)

**Comment by gregchak** [comment]
That's good.  That's exactly what you should see.  So now use that base URL along with the token you used in the browser for the import tool.

**Comment by magnumforce2006** [comment]
I'm already using 

http://localhost:32400/

As the server URL. Along with the same token

And still getting the error.

Could this have anything to do with my playlist, or is it definitely a connection issue?

**Comment by Stryp0** [comment]
I am having the same issue. Server is responding correctly from the URL when viewed in browser. Copying that same URL gives me an error:

2024-01-10 20:32:53.743 +01:00 [ERR] Errors occurred while processing request. System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Exception: The text associated with this error code could not be found.

**Comment by zbourdeau** [comment]
I am having the same issue. URL loads fine in the browser as well so I don't think it's a firewalling problem. I am running on windows server 2022 so I am wondering whether it's something specific to that OS.

Was there any update on how to workaround this issue?

**Comment by zbourdeau** [comment]
It looks like the endpoint is throwing a http 500 error underneath the covers. Has the upload endpoint been changed? I am curious whether this is just limited to my configuration.

{StatusCode: 500, ReasonPhrase: 'Internal Server Error', Version: 1.1, Content: System.Net.Http.HttpConnectionResponseContent, Headers:
{
  X-Plex-Protocol: 1.0
  Connection: Keep-Alive
  Keep-Alive: timeout=20
  Cache-Control: no-cache
  Date: Sat, 07 Dec 2024 19:12:18 GMT
  Content-Length: 109
  Content-Type: text/html
}}

**Comment by zbourdeau** [comment]
I found that the playlist's contents needs to exactly match the song in order to import properly. If you receive a http 500 error you likely have a playlist problem and should check the paths. In my case it was a case sensitive folder path for the .mp3 file.
